listener/upload.go

```diff
@@ -121,46 +136,52 @@ func processUpload(hostID string, account string, identity string) error {
 	// Create new context, which has the apikey value set. This is then used as a value for `x-rh-identity`
 	ctx := context.WithValue(context.Background(), inventory.ContextAPIKey, apiKey)
 
-	inventoryData, _, err := inventoryClient.HostsApi.ApiHostGetHostSystemProfileById(ctx, []string{hostID}, nil)
+	hostResults, _, err := inventoryClient.HostsApi.ApiHostGetHostById(ctx, []string{hostID}, nil)
+	if err != nil {
+		return errors.Wrap(err, "could not query inventory")
+	}
+
+	profileResults, _, err := inventoryClient.HostsApi.ApiHostGetHostSystemProfileById(ctx, []string{hostID}, nil)
 	if err != nil {
 		return errors.Wrap(err, "could not inventory system profile")
 	}
 
 	utils.Log().Debug("System profile download complete")
 
-	if inventoryData.Count == 0 {
-		return errors.Wrap(err, "no system details returned, host is probably deleted")
+	if profileResults.Count == 0 || hostResults.Count == 0 {
```

Comment:
It's easier to debug code without these logical combinations. It's better to throw separate errors for each case. Moreover you can check `hostResults.Count` immediately after this var is obtained and not to call inventory API next time if it's zero.